tox.ini

```diff
@@ -1,76 +1,147 @@
-    astropy31: astropy<3.2
+
+    # The devdeps factor is intended to be used to install the latest developer version.
+    # of key dependencies.
+    devdeps: cython
+    devdeps,figure_astropydev: git+https://github.com/astropy/astropy
+    devdeps: astropy<3.2
+
+    # Just installs the extra pytest-cov plugin.
+    cov: pytest-cov
+
+    # These are specfici online extras we use to run the online tests.
     online: pytest-rerunfailures
     online: pytest-timeout
     online: astroquery
-    astropydev,build_docs: reproject
-    astropydev,build_docs: astroquery
-    astropydev,build_docs: jplephem
+
+# The following indicates which extras_require from setup.cfg will be installed
+# dev is special in that it installs everything
+extras =
+    all,tests
+
 commands =
-    offline,astropydev,astropy31: {env:PYTEST_COMMAND} {posargs}
+    pip freeze
+    !online: {env:PYTEST_COMMAND} {posargs}
     online: {env:PYTEST_COMMAND} --reruns 2 --timeout=180 --remote-data=any {posargs}
     astropydev: bash -ec "for f in ../examples/*/*.py; do [[ $f == *skip* ]] && continue; python "$f"; done"
 
 [testenv:build_docs]
-changedir = {toxinidir}
-extras = dev
-setenv =
-    HOME = {envtmpdir}
-commands = sphinx-build docs docs/_build/html -W -b html -d docs/_build/.doctrees {posargs}
+changedir = docs
+description = Invoke sphinx-build to build the HTML docs
+extras = all,docs
+commands =
+    pip freeze
+    sphinx-build . ./_build/html -W -b html -d .docs/_build/.doctrees {posargs}
+
+[testenv:codestyle]
+# We list the warnings/errors to check for here rather than in setup.cfg because
+# we don't want these options to apply whenever anyone calls flake8 from the
+# command-line or their code editor - in this case all warnings/errors should be
+# checked for. The warnings/errors we check for here are:
+# E101 - mix of tabs and spaces
+# W191 - use of tabs
+# W291 - trailing whitespace
+# W292 - no newline at end of file
+# W293 - trailing whitespace
+# W391 - blank line at end of file
+# E111 - 4 spaces per indentation level
+# E112 - 4 spaces per indentation level
+# E113 - 4 spaces per indentation level
+# E301 - expected 1 blank line, found 0
+# E302 - expected 2 blank lines, found 0
+# E303 - too many blank lines (3)
+# E304 - blank lines found after function decorator
+# E305 - expected 2 blank lines after class or function definition
+# E306 - expected 1 blank line before a nested definition
+# E502 - the backslash is redundant between brackets
+# E722 - do not use bare except
+# E901 - SyntaxError or IndentationError
+# E902 - IOError
+# E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree
+# F822: undefined name in __all__
+# F823: local variable name referenced before assignment
+skip_install = true
+description = check code style, e.g. with flake8
+deps = flake8
+commands = flake8 sunpy --count --select=E101,W191,W291,W292,W293,W391,E111,E112,E113,E30,E502,E722,E901,E902,E999,F822,F823
```

Comment:
Should we limit the pre-commit flake 8 to this list of checks?

Reply:
I mean I can? I thought the idea was that we check a subset on the CI but make IDE users sort all problems?

Reply:
I like the idea of having the pre-commit be a way to make the style CI pass.

Reply:
Already done.